modin/core/storage_formats/pandas/query_compiler.py

```diff
@@ -2477,12 +2477,6 @@ def quantile_builder(df, **kwargs):
         result = self.__constructor__(new_modin_frame)
         return result.transpose() if axis == 1 else result
 
-    def query(self, expr, **kwargs):
-        def query_builder(df, **modin_internal_kwargs):
-            return df.query(expr, inplace=False, **kwargs, **modin_internal_kwargs)
-
-        return self.__constructor__(self._modin_frame.filter(1, query_builder))
```

Comment:
After this PR we see about ~1.6x perf degradation for one of our internal workloads that uses simple queries a lot.

@mvashishtha can you please check #6499?